internal/controller/pkg/resolver/reconciler.go

```diff
@@ -375,3 +401,95 @@ func (r *Reconciler) findDependencyVersion(ctx context.Context, dep *v1beta1.Dep
 
 	return addVer, nil
 }
+
+// UpdatableVersionFinder is an implementation of versionFinder that allows for version upgrade/downgrade capability considering parent constraints.
+type UpdatableVersionFinder struct {
```

Comment:
```suggestion
type UpgradingVersionFinder struct {
```

or

```suggestion
type UpdatingVersionFinder struct {
```

nit: version finder itself is not updatable, it is an updating type of a version finder.